presto-main/src/main/java/com/facebook/presto/type/TypeUtils.java

```diff
@@ -85,7 +85,8 @@ else if (!type.getJavaType().isPrimitive()) {
             }
         }
         catch (Throwable throwable) {
-            throw Throwables.propagate(throwable);
+            throwIfUnchecked(throwable);
```

Comment:
There are a lot of places through the code where interruption flag could be swallowed. Most of the places are rather just don't matter, but for example here. This code invokes some method handle, that hypothetically can call some of the methods that throw `InterruptedException`. And this piece of code can be run in a Driver, that we are trying to interrupt (https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/Driver.java#L793). So i don't understand what places do we need to handle, and what places we don't? And why don't just always handle it to be safe?  

I understand that if no exception is ever caught the thread is gonna finish anyway, no matter what the exception is (InterruptedException or RuntimeException). 

But in some places we do interrupted specific logic, like here:

https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/Driver.java#L556

Guava also suggest to use

```
catch (Exception e) {
  throwIfUnchecked(e);
  throw new RuntimeException(e);
}
```
as a silver bullet, and there is nothing about interruption flag recovery.

Reply:
This also bothers me from a theoretical and code style perspective. However, in practice, this is a hash code operator, and we'd be adding extra code in case someone does this:
```java
@ScalarOperator(HASH_CODE)
@SqlType(StandardTypes.BIGINT)
public static long hashCode(@SqlType(StandardTypes.IPADDRESS) Slice value)
        throws InterruptedException
{
    Thread.sleep(10); // WHY????
    return XxHash64.hash(value);
}
```
So we do want to add extra to handle this case?

BTW, this method `hashPosition()` shouldn't exist. It's only used in `MapHashCodeOperator` which should do the type specialization as part of its own specialization. Doing it for every single map entry is horrible. So I'd fix this by deleting the method.

Reply:
I looked through the code carefully, and i think i was exaggerating by saying `There are a lot of places`. There are may places, but most of them are either in Tests, or in Guice bootstrapping.

However i found one more place that seems to deserve attention:

https://github.com/prestodb/presto/pull/10490/files#diff-d014632c35cc5d87beeedecdfe1827efR163

Reply:
I just looked into that. It's a factory for creating state serializers, so the same reasoning seems to apply.